apps/openmw/mwinput/bindingsmanager.cpp

```diff
@@ -0,0 +1,738 @@
+        descriptions[A_QuickKey8] = "sQuick8Cmd";
+        descriptions[A_QuickKey9] = "sQuick9Cmd";
+        descriptions[A_QuickKey10] = "sQuick10Cmd";
+        descriptions[A_AlwaysRun] = "sAlways_Run";
+        descriptions[A_QuickSave] = "sQuickSaveCmd";
+        descriptions[A_QuickLoad] = "sQuickLoadCmd";
+
+        if (descriptions[action].empty())
+            return std::string(); // not configurable
+
+        return "#{" + descriptions[action] + "}";
+    }
+
+    std::string BindingsManager::getActionKeyBindingName(int action)
+    {
+        if (mInputBinder->getChannel(action)->getControlsCount() == 0)
+            return "#{sNone}";
+
+        ICS::Control* c = mInputBinder->getChannel(action)->getAttachedControls().front().control;
+
+        SDL_Scancode key = mInputBinder->getKeyBinding(c, ICS::Control::INCREASE);
+        unsigned int mouse = mInputBinder->getMouseButtonBinding(c, ICS::Control::INCREASE);
+        ICS::InputControlSystem::MouseWheelClick wheel = mInputBinder->getMouseWheelBinding(c, ICS::Control::INCREASE);
+        if (key != SDL_SCANCODE_UNKNOWN)
+            return MyGUI::TextIterator::toTagsString(mInputBinder->scancodeToString(key));
+        else if (mouse != ICS_MAX_DEVICE_BUTTONS)
+            return "#{sMouse} " + std::to_string(mouse);
+        else if (wheel != ICS::InputControlSystem::MouseWheelClick::UNASSIGNED)
+            switch (wheel)
+            {
+                case ICS::InputControlSystem::MouseWheelClick::UP:
+                    return "Mouse Wheel Up";
+                case ICS::InputControlSystem::MouseWheelClick::DOWN:
+                    return "Mouse Wheel Down";
+                case ICS::InputControlSystem::MouseWheelClick::RIGHT:
+                    return "Mouse Wheel Right";
+                case ICS::InputControlSystem::MouseWheelClick::LEFT:
+                    return "Mouse Wheel Left";
+                default:
+                    return "#{sNone}";
+            }
+        else
+            return "#{sNone}";
+    }
+
+    std::string BindingsManager::getActionControllerBindingName(int action)
+    {
+        if (mInputBinder->getChannel(action)->getControlsCount() == 0)
+            return "#{sNone}";
+
+        ICS::Control* c = mInputBinder->getChannel(action)->getAttachedControls().front().control;
+
+        if (mInputBinder->getJoystickAxisBinding(c, sFakeDeviceId, ICS::Control::INCREASE) != ICS::InputControlSystem::UNASSIGNED)
+            return sdlControllerAxisToString(mInputBinder->getJoystickAxisBinding(c, sFakeDeviceId, ICS::Control::INCREASE));
+        else if (mInputBinder->getJoystickButtonBinding(c, sFakeDeviceId, ICS::Control::INCREASE) != ICS_MAX_DEVICE_BUTTONS)
+            return sdlControllerButtonToString(mInputBinder->getJoystickButtonBinding(c, sFakeDeviceId, ICS::Control::INCREASE));
+        else
+            return "#{sNone}";
+    }
+
+    std::vector<int> BindingsManager::getActionKeySorting()
```

Comment:
This vector and controller vector below can be initialized straightaway, and they only need to be initialized once. That should reduce some overhead and save some lines of code.